course_discovery/settings/base.py

```diff
@@ -440,102 +446,35 @@
     'DOC_EXPANSION': 'list',
 }
 
-# Elasticsearch uses index settings to specify available analyzers.
-# We are adding the lowercase analyzer and tweaking the ngram analyzers here,
-# so we need to use these settings rather than the index defaults.
-# We are making these changes to enable autocomplete for the typeahead endpoint.
```

Comment:
Do we have a replacement for the lowercase / ngram tweaks here?

Reply:
Yes of course. Since we now describe an index for each model exactly, we indicate, if necessary, for each field which analyzer should be used. If this is necessary, of course, since for some fields it is not necessary to explicitly indicate, because what is by default is quite enough. I also wrote my analyzers where needed. (here is the module, please pay attention https://github.com/raccoongang/course-discovery/blob/haystack-migration-to-dsl/course_discovery/apps/course_metadata/search_indexes/documents/analyzers.py).
In fact, we have the same thing as here `ELASTICSEARCH_INDEX_SETTINGS`, but done a little differently, i.e. in a modern way, so to speak.
Also, Michael, here you can see how it works for documents modulehttps://github.com/raccoongang/course-discovery/blob/haystack-migration-to-dsl/course_discovery/apps/course_metadata/search_indexes/documents/

And as a proof that we brake nothing, that all tests pass 🙂 And there are tests that take into account that the required analyzer must be in the context field.

Reply:
Ah makes sense, thanks!